tests/Unit/CallPowerTest.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace Tests\Unit;
+
+use Exception;
+use Tests\TestCase;
+use Illuminate\Support\Facades\Bus;
+use Chompy\Jobs\CreateCallPowerPostInRogue;
+use DoSomething\Gateway\Resources\NorthstarUser;
+
+class CallPowerTest extends TestCase
+{
+    /**
+     * Mock the getUser Northstar Call.
+     *
+     * @return user
+     */
+    public function mockGetUserNSCall()
+    {
+        $this->northstarMock->shouldReceive('getUser')->andReturnUsing(function ($type, $id) {
+            return new NorthstarUser([
+                    'id' => $type === 'id' ? $id : $this->faker->northstar_id,
+                    'first_name' => $this->faker->firstName,
+                    'last_name' => $this->faker->lastName,
+                    'birthdate' => $this->faker->date,
+                    'email' => $this->faker->email,
+                    'mobile' => $this->faker->phoneNumber,
+                ]);
+        });
+    }
+
+    /**
+     * Test that a post with a completed status successfully sends to Rogue.
+     *
+     * @return void
+     */
+    public function testCompletedCallStatus()
+    {
+        // Fake the job being dispatched.
+        Bus::fake();
+
+        // Create a post with completed as the status.
+        $response = $this->withChompyApiKey()->json('POST', 'api/v1/callpower/call', [
+            'mobile' => '1234567891',
+            'callpower_campaign_id' => '1',
+            'status' => 'completed',
+            'call_timestamp' => '2017-11-09 06:34:01.185035',
+            'call_duration' => 50,
+            'campaign_target_name' => 'Mickey Mouse',
+            'campaign_target_title' => 'Representative',
+            'campaign_target_district' => 'FL-7',
+            'callpower_campaign_name' => 'Test',
+            'number_dialed_into' => '+12028519273',
+        ]);
+
+        $response->assertSuccessful();
+
+        Bus::assertDispatched(CreateCallPowerPostInRogue::class);
```

Comment:
It miiiiiight actually make sense to remove the `Bus::fake()` call from this test (or create another "successful" test down below that directly dispatches the job like the failure ones do) so that we can test the "internals" of the job function as expected (for example, that this payload results in a particular set of Northstar & Rogue API calls).

Reply:
@DFurnes ok! If I remove the `Bus::fake()`, I would want to mock the NS and Rogue calls, right? But if I mock a successful Rogue call, would that be testing anything ?

Reply:
We'd be asserting the format of the user & post that gets created for the given CallPower payload. So, for example, that the given `call_power_campaign_id` results in a post with the right `action_id` getting created, etc.

Reply:
ah ok sorry I know you've explained this to me so many times! Updated this and the one below to not use `Bus::fake()` anymore!